craft_parts/plugins/properties.py

```diff
@@ -42,19 +39,38 @@ class PluginProperties(PluginPropertiesModel):
     build step is dirty. This can be overridden in each plugin if needed.
     """
 
+    model_config = pydantic.ConfigDict(
+        alias_generator=lambda s: s.replace("_", "-"),
+        extra="forbid",
+        frozen=True,
+        validate_assignment=True,
+    )
+
+    plugin: Any = ""
```

Comment:
why Any?

Reply:
It was left over from what @tigarmo suggests when I was playing around with different ways to make creating child nicer and cleaner with pyright. In the end I went with double-declaring `frozen=True` though, so I'll switch this back to `str`